crates/bevy_ecs/src/schedule/executor/multi_threaded.rs

```diff
@@ -218,67 +241,38 @@ impl SystemExecutor for MultiThreadedExecutor {
 
         let SyncUnsafeSchedule {
             systems,
-            mut conditions,
+            conditions,
         } = SyncUnsafeSchedule::new(schedule);
 
         ComputeTaskPool::get_or_init(TaskPool::default).scope_with_executor(
             false,
             thread_executor,
             |scope| {
-                // the executor itself is a `Send` future so that it can run
-                // alongside systems that claim the local thread
-                #[allow(unused_mut)]
-                let mut executor = Box::pin(async {
-                    let world_cell = world.as_unsafe_world_cell();
-                    while self.num_completed_systems < self.num_systems {
-                        // SAFETY:
-                        // - self.ready_systems does not contain running systems.
-                        // - `world_cell` has mutable access to the entire world.
-                        unsafe {
-                            self.spawn_system_tasks(scope, systems, &mut conditions, world_cell);
-                        }
-
-                        if self.num_running_systems > 0 {
-                            // wait for systems to complete
-                            if let Ok(result) = self.receiver.recv().await {
-                                self.finish_system_and_handle_dependents(result);
-                            } else {
-                                panic!("Channel closed unexpectedly!");
-                            }
-
-                            while let Ok(result) = self.receiver.try_recv() {
-                                self.finish_system_and_handle_dependents(result);
-                            }
-
-                            self.rebuild_active_access();
-                        }
-                    }
-                });
-
-                #[cfg(feature = "trace")]
-                let executor_span = info_span!("multithreaded executor");
-                #[cfg(feature = "trace")]
-                let mut executor = executor.instrument(executor_span);
-
-                // Immediately poll the task once to avoid the overhead of the executor
-                // and thread wake-up. Only spawn the task if the executor does not immediately
-                // terminate.
-                if block_on(poll_once(&mut executor)).is_none() {
-                    scope.spawn(executor);
-                }
+                let context = Context {
```

Comment:
Is it possible to merge this with SyncUnsafeSchedule? Both of which live for the duration of the execution.

Reply:
Yes, I think so.  I mainly left them separate to reduce the churn, and because I wasn't sure what to name things.  

A related issue is that `Context` gets copied into each task.  It holds a reference to the `scope`, so we have to construct the `Context` inside the closure passed to `scope_with_executor`, which means it doesn't live long enough to get passed to spawned tasks by reference.  

If we want to minimize the size of the tasks, then it would make sense to create one struct outside of the `scope_with_executor` call, so that we're only copying two pointers in each task.  

I'll try to come up with some better names and reorganize.  

(The other option is to get rid of the wrapping struct and pass `scope` as a separate reference, but then you need explicit lifetimes everywhere.)